TRV11 Flow Logs

iliyas-kiwimesh opened the pull request.

sharique827 commented:
## METRO_FLOW

{
            "on_search1": {
                "Tags_descriptor_code": "Route_info tag should not be present inside tags"
            },
            "search2": {
                "stops": "Fulfillment stops are missing or empty",
            },
            "on_search2":{
              "ietms_erroe":"items array is missing or is empty at providers0".
            },
            "select": {
                "tmpstmp": "Timestamp for /on_search api cannot be greater than or equal to /select api",
                "item[0].item_id": "/message/order/items/id in item: 6fa73d13-f062-4870-9acf-bde318ca2bc8 should be one of the /item/id mapped in previous call"
            },
            "on_select": {
                "item[0].item_id": "/message/order/items/id in item: 6fa73d13-f062-4870-9acf-bde318ca2bc8 should be one of the /item/id mapped in select"
            },
            "init": {
                "tmpstmp": "Timestamp for /on_search api cannot be greater than or equal to /init api",
                "item[0].item_id": "/message/order/items/id in item: 6fa73d13-f062-4870-9acf-bde318ca2bc8 should be one of the /item/id mapped in previous call"
            },
            "confirm": {
                "item[0].item_id": "/message/order/items/id in item: 6fa73d13-f062-4870-9acf-bde318ca2bc8 should be one of the /item/id mapped in previous call"
            },
        }
        
        
        @iliyas-kiwimesh / @atul-kiwimesh

sharique827 commented:
@iliyas-kiwimesh 
You have enetered search 1 payload in search2 file and search 2 payload in search 1 json file. 
Submit the payload correctly with their file name

iliyas-kiwimesh commented:
@sharique827 
I have updated search flows and updated logs based on your above feedback. Request you to verify logs.
Thank you.

sharique827 commented:
@iliyas-kiwimesh 

TRV11

 - "Timestamp for /on_search1 api cannot be greater than or equal to /search2 api",

iliyas-kiwimesh commented:
@sharique827 Thank you for the feedback. We have updated the logs. Request you to help us on this.

sharique827 commented:
## BUS FLOW
{
  "search2": {
    "tmpstmp": "Timestamp for /on_search api cannot be greater than or equal to /search api",
  },
  "select": {
    "item[0].item_id": "/message/order/items/id in item: 6a41e30f-df98-4d1c-8b2f-faf49d190fa2 should be one of the /item/id mapped in previous call"
  },
  "soft_on_cancel": {
    "tmpstmp": "Timestamp for /soft_cancel api cannot be greater than or equal to /soft_on_cancel api",
  },
  "confirm_on_cancel": {
    "message_id": "Message Id for /confirm_cancel and /confirm_on_cancel api should be same.",
    "tmpstmp": "Timestamp for /confirm_cancel api cannot be greater than or equal to /confirm_on_cancel api",
  }
}

@iliyas-kiwimesh

iliyas-kiwimesh commented:
@iliyas-kiwimesh Pushed new logs, request you to help us with it.
Thank you.

iliyas-kiwimesh commented:
@sharique827 Request you to escalate review of this logs submission.
We are eager to move into prod. Thank you

sharique827 commented:
@iliyas-kiwimesh 
### BUS FLOW
"select": {
                "item[0].item_id": "/message/order/items/id in item: 6a41e30f-df98-4d1c-8b2f-faf49d190fa2 should be one of the /item/id mapped in previous call onsearch2"
            },

### Metro Flow
- verified

iliyas-kiwimesh commented:
@sharique827 We have updated logs, Request you to verify it.

sharique827 commented:
@iliyas-kiwimesh 
- BUS flow verified
- Get verified IGM and RSF logs also

iliyas-kiwimesh commented:
@sharique827 How to get verified IGM and RSF logs?

sharique827 commented:
@iliyas-kiwimesh 
Get that verified by Amit/Jyoti/Parneet

iliyas-kiwimesh commented:
@amit2311 Would you please verify IGM and RSF logs? Thank you.

iliyas-kiwimesh commented:
@sharique827 Would you please forward this verification to respective person?
Thank you.

sharique827 commented:
@iliyas-kiwimesh , Kindly wait

iliyas-kiwimesh commented:
@sharique827 Request you to escalate verification of IGM and RSF logs. Thank you.

JyotiSharma0711 commented:
@iliyas-kiwimesh I have reviewed the logs for RSF 1.0.0 separate flows for succesful and correction are missing please review this

https://docs.google.com/document/d/1tx86sypacIRXgL9nlNBdvHz7cYQjQoyC/edit

iliyas-kiwimesh commented:
@JyotiSharma0711 We have added correction flow as well, request you to verify logs. Thank you.

JyotiSharma0711 commented:
### `receiver_recon` Validation Errors

* **Context Validation:**

  * `/context/domain` must be equal to one of the allowed values: `ONDC:NTS10`
  * `/context/core_version` must be equal to one of the allowed values: `1.0.0`

* **Orderbook Validation:**

  * `/message/orderbook/orders` must contain **at least 5 items**
  * `/message/orderbook/orders/0/state` must be one of: `Completed`

* **Payment Object Validation (for first order):**

  * `/message/orderbook/orders/0/payment/uri` must match the `uri` format
  * `/message/orderbook/orders/0/payment/tl_method` must be one of: `http/get`
  * `/message/orderbook/orders/0/payment/params/transaction_status` must be one of: `PAID`
  * `/message/orderbook/orders/0/payment/params/amount` must be a **number**
  * `/message/orderbook/orders/0/payment/type` must be one of: `ON-ORDER`
  * `/message/orderbook/orders/0/payment/status` must be one of: `PAID`

* **ONDC Extension Attributes Validation:**

  * `@ondc/org/buyer_app_finder_fee_type` must be one of: `Percentage`, `amount`
  * `@ondc/org/withholding_amount_status` must be one of: `Assert`
  * `@ondc/org/return_window` must match format: `duration`
  * `@ondc/org/return_window_status` must be one of: `Assert`
  * `@ondc/org/settlement_basis` must be one of: `Collection`
  * `@ondc/org/settlement_window` must match format: `duration`

* **Settlement Details Validation:**

  * `@ondc/org/settlement_details/0/settlement_counterparty` must be one of: `buyer-app`, `seller-app`
  * `@ondc/org/settlement_details/0/settlement_phase` must be one of: `sale-amount`, `withholding-amount`, `refund`
  * `@ondc/org/settlement_details/0/settlement_type` must be one of: `upi`, `neft`, `rtgs`

* **Order Recon Status:**

  * `/message/orderbook/orders/0/order_recon_status` must be one of: `01`

* **Timestamp Validation:**

  * `settlement_timestamp` should be **less than** both `created_at` and `updated_at`

---

### `on_receiver_recon` Validation Errors

* **Context Validation:**

  * `/context/domain` must be equal to one of the allowed values: `ONDC:NTS10`
  * `/context/core_version` must be equal to one of the allowed values: `1.0.0`

Refer to enums given in the contract

https://github.com/ONDC-Official/protocol-network-extension/blob/main/specifications/rsp/enums/recon_status.json

iliyas-kiwimesh commented:
Orderbook Validation:

/message/orderbook/orders/0/state must be one of: Completed

Payment Object Validation:

/message/orderbook/orders/0/payment/tl_method must be one of: http/get
/message/orderbook/orders/0/payment/params/transaction_status must be one of: PAID
/message/orderbook/orders/0/payment/status must be one of: PAID

ONDC Extension Attributes Validation:

@ondc/org/withholding_amount_status must be one of: Assert
@ondc/org/settlement_window must match format: duration

Settlement Details Validation:

@ondc/org/settlement_details/0/settlement_counterparty must be one of: buyer-app, seller-app
@ondc/org/settlement_details/0/settlement_phase must be one of: sale-amount, withholding-amount, refund
@ondc/org/settlement_details/0/settlement_type must be one of: upi, neft, rtgs

Order Recon Status:

/message/orderbook/orders/0/order_recon_status must be one of: 01


Above validations are as expected in our logs already, rest of the changes have been updated. @JyotiSharma0711  Request you to verify logs. Thank you.

amitsinha07 commented:
@iliyas-kiwimesh

You can **only close an issue** when it is **resolved from the seller's side**.

JyotiSharma0711 commented:
## Success Flow

### Receiver Reconciliation Errors

#### Schema Validation Errors
- **Invalid Bank Account Number Format**:  
  `settlement_bank_account_no` must match the pattern: `^[0-9]+$`  
  Path:  
  `/message/orderbook/orders/4/payment/@ondc~1org~1settlement_details/0/settlement_bank_account_no`

- **Invalid IFSC Code Format**:  
  `settlement_ifsc_code` must match the pattern: `^[A-Z]{4}0[A-Z0-9]{6}$`  
  Path:  
  `/message/orderbook/orders/4/payment/@ondc~1org~1settlement_details/0/settlement_ifsc_code`

#### Response Timestamp Validation
- **Invalid Timestamp**:  
  `context.timestamp` should be **greater than** `created_at`

---

## Correction Flow

### Receiver Reconciliation

- **Validation Status**:  
  Previous validation errors **are not resolved** in this flow.

iliyas-kiwimesh commented:
@amitsinha07 We have updated IGM logs with RESOLVED status in respondent actions. Thank you for the feedback.

@JyotiSharma0711 We have added bank details by replacing masked one's. We have checked timestamp it is as expected.

amitsinha07 commented:
### @iliyas-kiwimesh

**issue_close**  
`issue/created_at` should be equal to the `created_at` of the first issue.

iliyas-kiwimesh commented:
@amitsinha07  Thank you.
@JyotiSharma0711 Would you please verify RSF logs? Looking forward to move into production.

JyotiSharma0711 commented:
I'll revert you today over this

JyotiSharma0711 commented:
Can we connect today

JyotiSharma0711 commented:
As discussed please update your logs accordingly

iliyas-kiwimesh commented:
@JyotiSharma0711 Updated RSF logs, would you please check them? Thank you🙌

iliyas-kiwimesh commented:
<img width="1470" alt="Screenshot 2025-05-31 at 15 43 53" src="https://github.com/user-attachments/assets/925471fc-cb29-4c45-a66a-6bf246ec495f" />

@JyotiSharma0711/ @sharique827  Why this is stuck at search 2 in pramaan? search 1 is success. Would you please share more info on this search 2 pramaan does? Also @JyotiSharma0711 would you please verify RSF Logs which were updated based on your feedback? 
Thank you

JyotiSharma0711 commented:
**flow 1, correction_flow**

- invoice_no is there on on_receiver_recon but it's not there in receiver_recon

JyotiSharma0711 commented:
For Pramaan issue you can raise it here https://github.com/ONDC-Official/pramaan/issues
or drop a mail on techsupport@ondc.org

iliyas-kiwimesh commented:
@JyotiSharma0711 we have updated logs with invoice_no, please check them. Thank you.

JyotiSharma0711 commented:
Hi @iliyas-kiwimesh I can see the previous pointed error are still there in your PR 

Correction flow

receiever_recon

Orderbook Validation:

/message/orderbook/orders/0/state must be one of: Completed

Payment Object Validation:

/message/orderbook/orders/0/payment/tl_method must be one of: http/get
/message/orderbook/orders/0/payment/params/transaction_status must be one of: PAID
/message/orderbook/orders/0/payment/status must be one of: PAID

JyotiSharma0711 commented:
Logs approved for RSF 1.0.0

iliyas-kiwimesh commented:
@sharique827, @amitsinha07, @JyotiSharma0711, @92shreyansh Bus, Metro, IGM, RSF logs are approved. Thank you for all the feedback, Would you please merge this PR? We need to add github links when moving to prod.

iliyas-kiwimesh commented:
@sharique827, @JyotiSharma0711 Request to escalate merging of this PR. Thank you

JyotiSharma0711 commented:
@iliyas-kiwimesh Please wait notifiying the team
